tests/unittests/CoreGraphics.drawing/CGContextDrawing_GradientTests.cpp

```diff
@@ -0,0 +1,179 @@
+                                CGFloat components[],
+                                CGFloat locations[],
+                                CGGradientDrawingOptions options) {
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    CGGradientRef gradient =
+        CGGradientCreateWithColorComponents(colorspace, components, locations, std::extent<decltype(locations)>::value);
+
+    CGContextDrawLinearGradient(context, gradient, startPoint, endPoint, options);
+    CFRelease(colorspace);
+    CFRelease(gradient);
+}
+
+static void _drawShortLinearGradientWithOptions(CGContextRef context, CGRect bounds, CGGradientDrawingOptions option) {
+    CGFloat locations[2] = { 0, 1 };
+    CGFloat components[8] = { 0.0, 1, 0.0, 1.0, 1.0, 0, 0, 1.0 };
+
+    CGRect borderRect = CGRectInset(bounds, 30, 50);
+
+    _drawLinearGradient(context,
+                        borderRect.origin,
+                        CGPointMake(borderRect.size.width, borderRect.size.height),
+                        components,
+                        locations,
+                        option);
+}
+
+DISABLED_DRAW_TEST_F(CGGradient, LinearGradient, UIKitMimicTest) {
+    CGFloat locations[2] = { 0, 1 };
+    CGFloat components[8] = { 0.0, 0.0, 1, 1.0, 1.0, 0, 0, 1.0 };
+    _drawLinearGradient(GetDrawingContext(),
+                        CGPointMake(0, 0),
+                        CGPointMake(512, 1024),
+                        components,
+                        locations,
+                        kCGGradientDrawsBeforeStartLocation);
+}
+
+DISABLED_DRAW_TEST_F(CGGradient, LinearGradientShortBothSides_Options_0, UIKitMimicTest) {
+    _drawShortLinearGradientWithOptions(GetDrawingContext(), GetDrawingBounds(), 0);
+}
+
+DISABLED_DRAW_TEST_F(CGGradient, LinearGradientShortBothSides_Options_kCGGradientDrawsBeforeStartLocation, UIKitMimicTest) {
+    _drawShortLinearGradientWithOptions(GetDrawingContext(), GetDrawingBounds(), kCGGradientDrawsBeforeStartLocation);
+}
+
+DISABLED_DRAW_TEST_F(CGGradient, LinearGradientShortBothSides_Options_kCGGradientDrawsAfterEndLocation, UIKitMimicTest) {
+    _drawShortLinearGradientWithOptions(GetDrawingContext(), GetDrawingBounds(), kCGGradientDrawsAfterEndLocation);
+}
+
+DRAW_TEST_F(CGGradient, LinearGradientInvalidCount, UIKitMimicTest) {
+    CGFloat locations[] = { 0.0 };
+
+    CGFloat components[] = {
+        0.85, 0, 0, 1.0,
+    };
+
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    CGGradientRef gradient = CGGradientCreateWithColorComponents(colorspace, components, locations, 0);
+
+    CGContextDrawLinearGradient(GetDrawingContext(),
+                                gradient,
+                                CGPointMake(0, 0),
+                                CGPointMake(512, 1024),
+                                kCGGradientDrawsBeforeStartLocation);
+    CFRelease(colorspace);
+    CFRelease(gradient);
+}
+
+DISABLED_DRAW_TEST_F(CGGradient, LinearGradient2, UIKitMimicTest) {
```

Comment:
uh it's going to be a huge wrapped convoluted function, going to lock it down on this. 

---
In reply to: [93496632](https://github.com/Microsoft/WinObjC/pull/1535#discussion_r93496632) [](ancestors = 93496632)